### PR TITLE
Make 'unknown signing intent' message more helpful

### PR DESCRIPTION
**Pull request by twaugh**
Include valid names in the error message.

Signed-off-by: Tim Waugh <twaugh@redhat.com>

### CODE REVIEW

File: atomic_reactor/plugins/pre_reactor_config.py [hunk 1]
```diff
@@ -438,11 +438,17 @@ def __init__(self, signing_intents, default_signing_intent):
...
+            this_name = entry['name']
+            if this_name == name:
                 return entry
 
-        raise ValueError('unknown signing intent name "{}"'.format(name))
+            valid.append(this_name)
+
+        raise ValueError('unknown signing intent name "{}", valid names: {}'
+                         .format(name,
+                                 ', '.join(valid)))
```
Comment (mlangsdorf) [inline atomic_reactor/plugins/pre_reactor_config.py]: Nit: can we get the format on a single line?

Comment (twaugh) [reply atomic_reactor/plugins/pre_reactor_config.py]: Updated.

File: atomic_reactor/plugins/pre_reactor_config.py [hunk 2]
```diff
@@ -438,11 +438,16 @@ def __init__(self, signing_intents, default_signing_intent):
...
     def get_signing_intent_by_name(self, name):
+        valid = []
         for entry in self.signing_intents:
-            if entry['name'] == name:
+            this_name = entry['name']
+            if this_name == name:
                 return entry
 
-        raise ValueError('unknown signing intent name "{}"'.format(name))
+            valid.append(this_name)
```
Comment (lcarva) [inline atomic_reactor/plugins/pre_reactor_config.py]: Neat!

**Comment by lcarva** [comment]
Updated release notes draft:
> - 'unknown signing intent' error message states possible valid values

**Pull request merged**